pkg/graphviz/traceflow_test.go

```diff
@@ -181,6 +192,152 @@ func TestGenGraph(t *testing.T) {
+					Flags:    2,
+					Protocol: 1,
+					TTL:      63,
+				},
+				TransportHeader: crdv1alpha1.TransportHeader{
+					UDP: &crdv1alpha1.UDPHeader{
+						SrcPort: 68,
+						DstPort: 80,
+					},
+				},
+				Length: 84,
+				SrcIP:  "10.10.1.4",
+			},
+			Phase: crdv1alpha1.Succeeded,
+			Results: []crdv1alpha1.NodeResult{
+				{
+					Node: "k8s-node-worker-1",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:    crdv1alpha1.ActionForwarded,
+							Component: crdv1alpha1.ComponentSpoofGuard,
+						},
+						{
+							Action:        crdv1alpha1.ActionForwarded,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+							TunnelDstIP:   "192.168.77.100",
+						},
+					},
+				},
+				{
+					Node: "k8s-node-control-plane",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:        crdv1alpha1.ActionReceived,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Classification",
+						},
+						{
+							Action:        crdv1alpha1.ActionDropped,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	nonliveTf := crdv1alpha1.Traceflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-non-live-traceflow",
+		},
+		Spec: crdv1alpha1.TraceflowSpec{
+			LiveTraffic: false,
+			Packet: crdv1alpha1.Packet{
+				IPHeader: crdv1alpha1.IPHeader{
+					Protocol: 1,
+				},
+				TransportHeader: crdv1alpha1.TransportHeader{
+					ICMP: &crdv1alpha1.ICMPEchoRequestHeader{
+						ID:       123,
+						Sequence: 1,
+					},
+				},
+			},
+			Source: crdv1alpha1.Source{
+				Namespace: "default",
+				Pod:       "pod-1",
+			},
+			Destination: crdv1alpha1.Destination{
+				Namespace: "default",
+				Pod:       "pod-2",
+			},
+		},
+		Status: crdv1alpha1.TraceflowStatus{
+			CapturedPacket: &crdv1alpha1.Packet{
+				DstIP: "10.10.0.2",
+				IPHeader: crdv1alpha1.IPHeader{
+					Flags:    2,
+					Protocol: 1,
+					TTL:      63,
+				},
+				Length: 84,
+			},
+			Phase: crdv1alpha1.Succeeded,
+			Results: []crdv1alpha1.NodeResult{
+				{
+					Node: "k8s-node-worker-1",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:    crdv1alpha1.ActionForwarded,
+							Component: crdv1alpha1.ComponentSpoofGuard,
+						},
+						{
+							Action:        crdv1alpha1.ActionForwarded,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+							TunnelDstIP:   "192.168.77.100",
+						},
+					},
+				},
+				{
+					Node: "k8s-node-control-plane",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:        crdv1alpha1.ActionReceived,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Classification",
+						},
+						{
+							Action:        crdv1alpha1.ActionDropped,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
```

Comment:
`ActionDropped` is used with ComponentNetworkPolicy. So, you can change Component and ComponentInfo accordingly. Also, you can add NetworkPolicy observation on source Node of this traceflow.